File: 2033. Minimum Operations to Make a Uni-Value Grid.java
class Solution {
    public int minOperations(int[][] grid, int x) {
        ArrayList<Integer> arr = new ArrayList<>();
        for(int i = 0; i < grid.length; i++){
            for(int j = 0; j < grid[0].length; j++){
                arr.add(grid[i][j]);
            }
        }

        Collections.sort(arr);
        int mid = arr.get(arr.size() / 2);
        int ans = 0;
        for(int i = 0; i < arr.size(); i++){
            if(arr.get(i) > mid){
                if(((double)(arr.get(i) - mid) / x) != (arr.get(i) - mid) / x) return -1;
                ans += (arr.get(i) - mid) / x;
            }
            else{
                if(((double)(mid - arr.get(i)) / x) != (mid - arr.get(i)) / x) return -1;
                ans += (mid - arr.get(i)) / x;
            }
        }

        return ans;
    }
}

File: 2033. Minimum Operations to Make a Uni-Value Grid.cpp
class Solution {
public:
    int minOperations(vector<vector<int>>& grid, int x) {
        vector<int> arr;
        for (const auto& row : grid) {
            for (int val : row) {
                arr.push_back(val);
            }
        }

        sort(arr.begin(), arr.end());
        int mid = arr[arr.size() / 2];
        int ans = 0;
        for (int val : arr) {
            int diff = val > mid ? val - mid : mid - val;
            if (diff % x != 0) return -1;
            ans += diff / x;
        }

        return ans;
    }
};